python/ray/includes/unique_ids.pxi

```diff
@@ -333,13 +334,13 @@ cdef class ClientActorRef(ActorID):
             raise TypeError("Unexpected type for id {}".format(id))
 
     def __dealloc__(self):
-        if client is None or client.ray is None:
-            # The client package or client.ray object might be set
-            # to None when the script exits. Should be safe to skip
-            # call_release in this case, since the client should have already
-            # disconnected at this point.
-            return
-        if client.ray.is_connected():
+        def _connected():
+            try:
+                return self._worker is not None and self._worker.is_connected()
+            except Exception:
```

Comment:
Avoid swallowing exceptions.

Reply:
Actually `is_connected()` only accesses some internal members. I think it's because this is `__dealloc__` from cython, so when we access self._worker, sometimes, the internal data has been cleaned up and it'll through attribute not found error.

I'm not an expert in cython, but I think we probably should use `def __del__` here so the worker's attribute will still be there. If it's not working I'll hide the exception into `is_connected()` method in worker.

Reply:
It seems `def __del__` hasn't been supported in cython before this https://github.com/cython/cython/pull/3804/files

I'll move try catch into `is_connected`